src/apps/testapps/testEdge.c

```diff
@@ -0,0 +1,298 @@
+        t_assert(H3_EXPORT(isValidEdge)(sf) == 0, "hexagons do not validate");
+
+        H3Index directedEdge = edge;
+        H3_SET_MODE(directedEdge, H3_DIRECTEDEDGE_MODE);
+        t_assert(H3_EXPORT(isValidEdge)(directedEdge) == 0,
+                 "directed edges do not validate");
+
+        H3Index hexagonWithReserved = sf;
+        H3_SET_RESERVED_BITS(hexagonWithReserved, 1);
+        t_assert(H3_EXPORT(isValidEdge)(hexagonWithReserved) == 0,
+                 "hexagons with reserved bits do not validate");
+
+        H3Index fakeEdge = sf;
+        H3_SET_MODE(fakeEdge, H3_EDGE_MODE);
+        t_assert(H3_EXPORT(isValidEdge)(fakeEdge) == 0,
+                 "edges without an edge specified don't work");
+        H3Index invalidEdge = sf;
+        H3_SET_MODE(invalidEdge, H3_EDGE_MODE);
+        H3_SET_RESERVED_BITS(invalidEdge, INVALID_DIGIT);
+        t_assert(H3_EXPORT(isValidEdge)(invalidEdge) == 0,
+                 "edges with an invalid edge specified don't work");
+
+        H3Index pentagon = 0x821c07fffffffff;
+        H3Index goodPentagonalEdge = pentagon;
+        H3_SET_MODE(goodPentagonalEdge, H3_EDGE_MODE);
+        H3_SET_RESERVED_BITS(goodPentagonalEdge, 2);
+        t_assert(H3_EXPORT(isValidEdge)(goodPentagonalEdge) == 1,
+                 "pentagonal edge validates");
+
+        H3Index badPentagonalEdge = goodPentagonalEdge;
+        H3_SET_RESERVED_BITS(badPentagonalEdge, 1);
+        t_assert(H3_EXPORT(isValidEdge)(badPentagonalEdge) == 0,
+                 "missing pentagonal edge does not validate");
+        // Case discovered by fuzzer that triggers pentagon deleted direction
+        // condition
+        H3Index badPentagonalEdge2 = 0x990d6700ff00ffff;
+        t_assert(H3_EXPORT(isValidEdge)(badPentagonalEdge2) == 0,
+                 "missing pentagonal edge 2 does not validate");
+
+        H3Index highBitEdge = edge;
+        H3_SET_HIGH_BIT(highBitEdge, 1);
+        t_assert(H3_EXPORT(isValidEdge)(highBitEdge) == 0,
+                 "high bit set edge does not validate");
+    }
+
+    TEST(cellToEdges) {
+        H3Index sf;
+        t_assertSuccess(H3_EXPORT(latLngToCell)(&sfGeo, 9, &sf));
+        H3Index edges[6] = {H3_NULL};
+        t_assertSuccess(H3_EXPORT(cellToEdges)(sf, edges));
+
+        for (int i = 0; i < 6; i++) {
+            t_assert(H3_EXPORT(isValidEdge)(edges[i]) == 1, "edge is an edge");
+            H3Index owner, destination;
+            owner = edges[i];
+            Direction dir = H3_GET_RESERVED_BITS(edges[i]);
+            H3_SET_MODE(owner, H3_CELL_MODE);
+            H3_SET_RESERVED_BITS(owner, 0);
+            int rotations = 0;
+            t_assertSuccess(
+                h3NeighborRotations(owner, dir, &rotations, &destination));
+            t_assert(owner == sf || destination == sf,
+                     "original cell is owner or neighbor");
+            t_assert(owner < destination, "owning cell sorts first");
```

Comment:
Nit: As above, I don't think this is worth a test, as it's an implementation detail

Reply:
Which cell is the owner and which is the non-owner seems to be part of the contract of undirected edge mode, though. That seems to need testing to ensure it is always adhered to. It can't really be changed after the fact, can it?